ordering/__init__.py

```diff
@@ -61,6 +61,18 @@ def compare(self, left_item: T, right_item: T) -> bool:
 
         return self._labels[left_item] < self._labels[right_item]
 
+    def replace(self, existing_item: T, new_item: T) -> None:
```

Comment:
The idea is essentially what I present on [one of my other branches](https://github.com/Yogi-B/python-order-maintenance/tree/bug/_Sentinel-leaking-into-public-interface), for a different reason.

Basically, have two sets of the core methods: a protected set with the actual implementation and minimal assertions; and a public set that performs any missing assertions, before delegating to its protected equivalent.